spatial-commands/src/main/java/org/codice/ddf/commands/spatial/geonames/GeoNamesCreateCommand.java

```diff
@@ -0,0 +1,195 @@
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileFilter;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.felix.gogo.commands.Argument;
+import org.apache.felix.gogo.commands.Command;
+import org.apache.karaf.shell.console.OsgiCommandSupport;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FSDirectory;
+import org.apache.lucene.util.Version;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Command(scope = "geonames", name = "create",
+        description = "Creates the local GeoNames index if it doesn't exist and overwrites it if " +
+                "it does.")
+public final class GeoNamesCreateCommand extends OsgiCommandSupport {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GeoNamesCreateCommand.class);
+
+    @Argument(index = 0, name = "file_location",
+            description = "Location of the GeoNames .txt file or directory containing GeoNames " +
+                    ".txt files.\n" +
+                    "If it's a directory, the contents of all the .txt files inside will be " +
+                    "added.\n" +
+                    "The files to add should be obtained from " +
+                    "http://download.geonames.org/export/dump and have the format described " +
+                    "there.", required = true)
+    private String inputLocation = null;
+
+    private static final PrintStream CONSOLE = System.out;
+
+    private static final String GEONAMES_INDEX = "geonames-index";
+
+    @Override
+    protected Object doExecute() {
+        final String inputFilesLocation = getInputFilesLocation(inputLocation);
+
+        if (inputFilesLocation != null) {
+            buildIndex(inputFilesLocation, true);
+        }
+
+        return null;
+    }
+
+    static String getInputFilesLocation(final String fileLocation) {
+        final File file = new File(fileLocation);
+        if (file.isDirectory() || FilenameUtils.isExtension(fileLocation, "txt")) {
+            return fileLocation;
+        }
+
+        CONSOLE.println("Input must be a .txt or a directory.");
+        return null;
+    }
+
+    static void buildIndex(final String inputFilesLocation, final boolean create) {
+        CONSOLE.println("Building index...");
+
+        Directory directory;
+
+        try {
+            directory = FSDirectory.open(new File(GEONAMES_INDEX));
+        } catch (IOException e) {
+            CONSOLE.println("Couldn't open the directory for the index.");
+            LOGGER.error("Error opening the directory for the GeoNames index", e);
```

Comment:
Can we specify where we want the index to be? I think the default should be in ${DDF_HOME}/data but it wouldn't hurt to allow an admin to specify an external directory either.